### PR TITLE
Dynatrace metrics exporter

### PR DESCRIPTION
**Pull request by dyladan**
**Description:** Add an exporter for metrics to Dynatrace

**Link to tracking Issue:** n/a

**Testing:** Tests are done for configuration and some serialization

**Documentation:** README.md added in `exporter/dynatraceexporter` which explains configuration and use of Dynatrace metric exporter.

### CODE REVIEW

**Comment by dyladan** [comment]
@kbrockhoff not sure who to tag here but you're assigned to this PR so I guess you're as good a person to ask as any. The tests seem to have failed in a completely unrelated part of the codebase (xray receiver). Is there a way you can force a rerun? Or should I just push another commit to trigger the build?

File: exporter/dynatraceexporter/serialization/serialization.go
Comment (anuraaga) [inline exporter/dynatraceexporter/serialization/serialization.go:80]: Think we need to handle `count == 0`

File: exporter/dynatraceexporter/serialization/serialization.go
Comment (anuraaga) [inline exporter/dynatraceexporter/serialization/serialization.go:103]: Ditto

File: exporter/dynatraceexporter/serialization/serialization.go
Comment (anuraaga) [inline exporter/dynatraceexporter/serialization/serialization.go:134]: Assuming it's safe to not have any escaping or anything of the value?

File: exporter/dynatraceexporter/metrics_exporter.go
Comment (dyladan) [reply exporter/dynatraceexporter/metrics_exporter.go:124]: I moved the validation while developing. you are correct it's validated

File: exporter/dynatraceexporter/serialization/serialization.go
Comment (dyladan) [reply exporter/dynatraceexporter/serialization/serialization.go]: Added escaping but I think the characters requiring escaping are not allowed by the spec anyways

**Comment by dyladan** [comment]
@anuraaga I addressed all of your comments except I am still building the string. I will try to get around to refactoring it to use a buffer but i don't know when that will be done and I dont think it should block this PR.

File: exporter/dynatraceexporter/metrics_exporter.go
Comment (dyladan) [reply exporter/dynatraceexporter/metrics_exporter.go]: I put this on my todo list but I don't think it should block this PR.

File: exporter/dynatraceexporter/LICENSE
Comment (bogdandrutu) [inline exporter/dynatraceexporter/LICENSE:5]: Why is this needed?

File: exporter/dynatraceexporter/config/config.go
Comment (bogdandrutu) [inline exporter/dynatraceexporter/config/config.go:29]: Do you want to use the confighttp.HttpClientSettigs from core?

**Review by bogdandrutu — COMMENTED** [review]
Please follow the instructions https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md#when-adding-a-new-component

File: exporter/dynatraceexporter/factory.go
Comment (kbrockhoff) [inline exporter/dynatraceexporter/factory.go:76]: Do you have plans to make these configurable in the future?

File: exporter/dynatraceexporter/metrics_exporter.go
Comment (kbrockhoff) [inline exporter/dynatraceexporter/metrics_exporter.go:70]: Consider handling processing for loop in a separate method.

File: exporter/dynatraceexporter/config/config.go
Comment (dyladan) [reply exporter/dynatraceexporter/config/config.go:29]: I wasn't aware it existed. I just looked at another exporter and did what it did. I think it was datadog? I can change to use that if you would prefer?

File: exporter/dynatraceexporter/LICENSE
Comment (dyladan) [reply exporter/dynatraceexporter/LICENSE:5]: I'll remove it, it was added by my client and I didn't notice.

**Comment by dyladan** [comment]
> Please follow the instructions [open-telemetry/opentelemetry-collector@`master`/CONTRIBUTING.md#when-adding-a-new-component](https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md#when-adding-a-new-component)

So would you like me to open a new PR with just the Readme, configuration, and factory?

File: exporter/dynatraceexporter/factory.go
Comment (dyladan) [reply exporter/dynatraceexporter/factory.go:76]: No. This is just what I saw another exporter do.

Comment (kbrockhoff) [reply exporter/dynatraceexporter/factory.go:81]: This is fine to start but as this gets deployed to different environments you will likely find this needs to be adjustable.

**Review by kbrockhoff — APPROVED** [review]

**Comment by dyladan** [comment]
What is the best way for me to resolve go.sum issues? Should I just re-run `go build` or should I use `go mod tidy`? `go mod tidy` removes the `memcachedreceiver` also for some reason.

File: exporter/dynatraceexporter/metrics_exporter.go
Comment (dyladan) [reply exporter/dynatraceexporter/metrics_exporter.go:72]: All that is done in the loop is some nil checking then it delegates to the correct serialization function. I could make the whole for loop body its own function but I think it unnecessary redirection while reading the code.

**Comment by dyladan** [comment]
I just wanted to check in to see what the current status of this is. I don't think this PR is waiting on me to do something but if it is please let me know. I'm not sure how many reviews are needed for this repo before a merge, but I think I've addressed all of the open PR comments. I have also removed the component from the components.go file as suggested by bogdan to limit the impact of merging this component until its stability is proven.

**Comment by tigrannajaryan** [comment]
@bogdandrutu do you have any other comments or good to merge?

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
